mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -780,12 +781,23 @@ public void updateAnnotations(MailboxPath mailboxPath, MailboxSession session, L
         for (MailboxAnnotation annotation : mailboxAnnotations) {
             if (annotation.isNil()) {
                 annotationMapper.deleteAnnotation(mailboxId, annotation.getKey());
-            } else {
+            } else if (isValid(mailboxId, annotation, annotationMapper)) {
                 annotationMapper.insertAnnotation(mailboxId, annotation);
             }
         }
     }
 
+    private boolean isValid(MailboxId mailboxId, MailboxAnnotation annotation, AnnotationMapper annotationMapper) throws AnnotationException {
+        if (annotation.size() > MailboxConstants.LIMIT_ANNOTATION_DATA) {
+            throw new AnnotationException("annotation too big.");
+        }
+        if (!annotationMapper.isExisted(mailboxId, annotation)
+            && annotationMapper.countAnnotations(mailboxId) >= MailboxConstants.LIMIT_ANNOTATIONS_ON_MAILBOX) {
+            throw new AnnotationException("too many annotations.");
```

Comment:
It's not capital letter cause it should be consistent output at MPT testing. No capital.
For the message, I followed the RFC at '4.3.  SETMETADATA Command'.